flink-streaming-connectors/flink-connector-elasticsearch2/src/main/java/org/apache/flink/streaming/connectors/elasticsearch2/ElasticsearchSink.java

```diff
@@ -177,7 +180,72 @@ public void open(Configuration configuration) {
 		if (LOG.isInfoEnabled()) {
 			LOG.info("Created Elasticsearch TransportClient {}", client);
 		}
+	}
+
+	@Override
+	public void invoke(final T element) {
+		ParameterTool params = ParameterTool.fromMap(userConfig);
+
+		if (params.has(CONFIG_NO_OF_CONN_RETRIES) && params.getInt(CONFIG_NO_OF_CONN_RETRIES) > 0) {
+			final Timer timer = new Timer(true);
+			TimerTask task = new TimerTask() {
+				@Override
+				public void run() {
+					// verify that we actually are connected to a cluster
+					ImmutableList<DiscoveryNode> nodes = ImmutableList.copyOf(((TransportClient) client).connectedNodes());
+					if (nodes.isEmpty()) {
+						if (LOG.isInfoEnabled()) {
+							LOG.info("Connection Lost..Trying to reconnect to Elasticsearch nodes...");
+						}
+						open(new Configuration());
```

Comment:
I would not recommend calling the open() method from invoke(). Open() is a "lifecycle" method called by Flink. You should assume its only called one.
However, you can move the (re)connect logic into a separate method.